index.html

```diff
@@ -631,8 +630,25 @@ <h2>DNS-Based Service Discovery</h2>
                         <dd>Type of the Thing Description, i.e. <code>Thing</code> or <code>Directory</code>.
                         If omitted, the type is assumed to be <code>Thing</code>.</dd>
                         <dt><code>scheme</code></dt>
-                        <dd>Scheme part of URL.  One of the following values can be specified, with the standard registered URI interpretations [[RFC7595]]: <code>http</code> (HTTP over TCP),<code>https</code> (HTTP over TLS/TCP), <code>coap</code> (CoAP over UDP), <code>coaps</code> CoAP over DTLS/UDP), <code>coap+tcp</code> (CoAP over TCP), or <code>coaps+tcp</code> (CoAP over TLS/TCP).
-                        If omitted, the scheme is assumed to be <code>http</code> for TCP and <code>coap</code> for UDP. </dd>
+                        <dd>Scheme part of URL.  One of the following values can be specified, with the standard registered URI interpretations [[RFC7595]]: <code>http</code> (HTTP over TCP), <code>https</code> (HTTP over TLS/TCP), <code>coap+tcp</code> (CoAP over TCP), or <code>coaps+tcp</code> (CoAP over TLS/TCP).
+                        If omitted, the scheme is assumed to be <code>http</code>.</dd>
+                    </dl>
+                </div>
+                <div class="rfc2119-assertion" id="introduction-dns-sd-txt-record-udp">
+                    <p>
+                    For UDP-based services,
+                    the following information MUST be included in the <code>TXT</code>
+                    record that is pointed to by the Service Instance Name:
+                    </p>
+                    <dl>
+                        <dt><code>td</code></dt>
+                        <dd>Absolute pathname of the Thing Description of the Thing or Thing Description of the <a>Thing Description Directory</a>.</dd>
+                        <dt><code>type</code></dt>
+                        <dd>Type of the Thing Description, i.e. <code>Thing</code> or <code>Directory</code>.
+                        If omitted, the type is assumed to be <code>Thing</code>.</dd>
+                        <dt><code>scheme</code></dt>
+                        <dd>Scheme part of URL.  One of the following values can be specified, with the standard registered URI interpretations [[RFC7595]]: <code>coap</code> (CoAP over UDP) or <code>coaps</code> (CoAP over DTLS/UDP).
+                        If omitted, the scheme is assumed to be <code>coap</code>.</dd>
```

Comment:
One thing I noticed when implementing DNS-SD discovery was that there is currently no way to include a port in the TXT record, so you have to rely on the default ports for the indicated/implied URI scheme. Would it make sense to add another optional attribute for the port here?

Reply:
Maybe I misunderstood the issue, but there is a dedicated entry for port that goes into the service registration. It doesn't need to be passed as a TXT record; see [here](https://www.rfc-editor.org/rfc/rfc6763#section-6) and other parts of the spec.

For example, see  https://github.com/TinyIoT/thing-directory/wiki/Discovery-with-DNS-SD (may not be compliant with the current Discovery spec but you can see the custom port).

The user manuals for registering a service via dns-sd and avahi CLI apps could also be used as reference:
- https://man.netbsd.org/dns-sd.1
- https://linux.die.net/man/1/avahi-publish-service